modules/social_features/social_activity/modules/social_activity_filter/src/Plugin/views/display/FilterBlock.php

```diff
@@ -0,0 +1,338 @@
+        'tags' => ['default' => 'tags'],
+      ],
+    ];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSettings(array $settings) {
+    $settings = parent::blockSettings($settings);
+
+    $settings['vocabulary'] = 'none';
+    $settings['tags'] = 'none';
+
+    return $settings;
+  }
+
+  /**
+   * Provide the summary for page options in the views UI.
+   *
+   * This output is returned as an array.
+   */
+  public function optionsSummary(&$categories, &$options) {
+    parent::optionsSummary($categories, $options);
+
+    if ($this->getOption('override_tags_filter')) {
+      $options['allow']['value'] .= ', ' . $this->t('Overridden Tags filter');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    if ($form_state->get('section') !== 'allow') {
+      return;
+    }
+    $customized_filters = $this->getOption('override_tags_filter');
+    $form['override_tags_filter'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Override Tags filters'),
+      '#description' => $this->t('Select the filters which users should be able to customize default values for when placing the views block into a layout.'),
+      '#default_value' => !empty($customized_filters) ? $customized_filters : [],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::submitOptionsForm($form, $form_state);
+
+    if ($form_state->get('section') === 'allow') {
+      $this->setOption('override_tags_filter', $form_state->getValue('override_tags_filter'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm(ViewsBlock $block, array &$form, FormStateInterface $form_state) {
+    parent::blockForm($block, $form, $form_state);
+
+    // Check if overridden filter option is enabled for current views block.
+    if (!$this->getOption('override_tags_filter')) {
+      return $form;
+    }
+
+    $allow_settings = $this->getOption('tags_filter');
+    $allow_settings += array_filter($this->getOption('allow'));
+    $block_configuration = $block->getConfiguration();
+
+    if ($vid = $form_state->get('new_options_tags')) {
+      $opt = $this->getTermOptionslist($vid);
+    }
+    else {
+      $opt = $this->getTermOptionslist($block_configuration['vocabulary']);
+    }
+
+    foreach ($allow_settings as $type => $enabled) {
+      if (empty($enabled)) {
+        continue;
+      }
+      switch ($type) {
+        case 'vocabulary':
+          $form['override']['vocabulary'] = [
+            '#type' => 'select',
+            '#title' => $this->t('Vocabulary'),
+            '#options' => $this->getVocabularyOptionsList(),
+            '#default_value' => $block_configuration['vocabulary'],
+            '#empty_option' => t('None'),
+            '#required' => TRUE,
+            '#ajax' => [
```

Comment:
@volodymyr-sydor-lemberg need to check if it works in all cases:

- Blocks library
- Dashboard
- Landing Page

Reply:
@ribel For blocks and dashboard it works good, only I added an additional condition to skip it for landing page

Reply:
done